lib/SILOptimizer/Differentiation/LinearMapInfo.cpp

```diff
@@ -139,6 +134,7 @@ LinearMapInfo::createBranchingTraceDecl(SILBasicBlock *originalBB,
   branchingTraceDecl->setImplicit();
   // Branching trace enums shall not be resilient.
   branchingTraceDecl->getAttrs().add(new (astCtx) FrozenAttr(/*implicit*/ true));
+  branchingTraceDecl->getAttrs().add(new (astCtx) UsableFromInlineAttr(/*Implicit*/ true));
```

Comment:
I think this should also respect the original linkage like line 90. We don't want to accidentally expose branching trace enums as ABI.

Reply:
Well. Here is the problem. They are explicitly non-resilient due to `@frozen`. Therefore they should be either public or `@usableFromInline@`.

Reply:
Not sure I understand. For a private or internal differentiable function, why should its derivative's branching trace enum be public or `@usableFromInline`?

Reply:
Why they are `@frozen` then? This attributes essentially implies that the type is public-ish. SIL verifier errors in case of non-public frozen type.

Reply:
> This attributes essentially implies that the type is public-ish.

Well, that's backwards...  It's a _bug_ that it's adding `@frozen` to non-ABI-public types.  The correct fix here would be to _not_ emit a `@frozen` attribute when it's not ABI-public.  Branching trace enums and linear map should have the exact same visibility and attributes.

Reply:
Fair enough. I though it was intentional due to comments and how the code was organized and thought there are some corner cases that I'm not aware of.

Will be fixed in the subsequent "tuples" patch :)